react/components/Table/index.js

```diff
@@ -26,11 +26,17 @@ class Table extends PureComponent {
       tableRowHeight: this.getRowHeight(props.density),
       selectedDensity: props.density,
       allChecked: false,
-      selectedRows: [],
+      selectedRows: this.props.bulkActions.selectedRows || [],
       allLinesSelected: false,
     }
   }
 
+  componentDidUpdate(prevProps) {
+    if(this.props.bulkActions && this.props.bulkActions.selectedRows) {
+      this.setState({selectedRows: this.props.bulkActions.selectedRows})
```

Comment:
Won't this call to `setState` in `componentDidUpdate` cause an endless loop? I'm feeling a lag when the bulkActions are used. Maybe this is caused by the fact that react limits the amount of renders. It wouldn't be stuck in an infinite loop but it would be the cause of this lag, since there's a lot of rerender happening.

Reply:
Me and @rafarubim have discussed and tested this part and we found that is not a problem. Furthermore, we add a new check(prop and state arrays are different) to prevent the component of re-rendering.